reframe/core/pipeline.py

```diff
@@ -1310,8 +1331,10 @@ def run(self):
         self.job.num_tasks_per_socket = self.num_tasks_per_socket
         self.job.num_cpus_per_task = self.num_cpus_per_task
         self.job.use_smt = self.use_multithreading
-        self.job.time_limit = self.time_limit
-
+        self.job.time_limit = (self.time_limit or rt.runtime().get_option(
+            f'systems/@{self.current_system.name}/partitions/'
```

Comment:
The loaded configuration has always only one system loaded:

```suggestion
            f'systems/0/partitions/'
```